File: Solutions/cpp/parkinglot/Vehicle.cc
#include "Vehicle.h"
#include <iostream>


Vehicle::Vehicle(std::string licensePlate, VehicleType type, std::string color)
    : licensePlate(licensePlate), type(type), color(color) {}


std::string Vehicle::getLicensePlate() const {
    return licensePlate;
}

VehicleType Vehicle::getType() const {
    return type;
}

std::string Vehicle::getColor() const {
    return color;
}

void Vehicle::displayInfo() const {
    std::cout << "Vehicle: " << color << " ";
    switch (type) {
        case VehicleType::CAR:
            std::cout << "Car";
            break;
        case VehicleType::MOTORCYCLE:
            std::cout << "Motorcycle";
            break;
        case VehicleType::TRUCK:
            std::cout << "Truck";
            break;
        case VehicleType::BUS:
            std::cout << "Bus";
            break;
        default:
            std::cout << "Unknown";
            break;
    }
    std::cout << " (License Plate: " << licensePlate << ")" << std::endl;
}

File: Solutions/golang/parkinglot/vehicle.go
package parkinglot

// VehicleType 定义车辆类型
type VehicleType int

const (
	CAR VehicleType = iota
	MOTORCYCLE
	TRUCK
)

// Vehicle 接口定义车辆的基本行为
type Vehicle interface {
	GetLicensePlate() string
	GetType() VehicleType
}

// BaseVehicle 提供Vehicle接口的基本实现
type BaseVehicle struct {
	licensePlate string
	vehicleType  VehicleType
}

func (v *BaseVehicle) GetLicensePlate() string {
	return v.licensePlate
}

func (v *BaseVehicle) GetType() VehicleType {
	return v.vehicleType
}